### PR TITLE
Fixed query widget pr

### PR DESCRIPTION
**Pull request by Steven-Eardley**
## This is still pointing to localhost!

PR for fixed query widget. to test: 
* run pip again to get new requirement
* run locally with DOAJ running on 5004
* check the page on /widgets renders properly
* copy the code from the **share | embed** button on /search and make a simple html page with it
* check the 'embedded' widget has the same results as your search

before deploying:
* bump version - there have been some JS changes to facetview
* **change http://locahost:5004 to https://doaj.org** in fixed_query_body.html **IMPORTANT**

the fixed query widget always will then only work via the live site - the js/css is hard coded to be retrieved from https://doaj.org.

The widgets page is accessed from sidebar link 'features' - renamed 'interacting with doaj'

### CODE REVIEW

**Comment by emanuil-tolev** [comment]
Thank you @Steven-Eardley :). Enjoy your holiday.

File: portality/static/doaj/js/doaj.facetview.theme.js [hunk 1]
```diff
@@ -87,6 +198,7 @@ function doajPager(options) {
      * options.page_size - number of results per page
      * options.data.found - the total number of records in the search result set
      */
+    console.log(options);
```
Comment (emanuil-tolev) [inline portality/static/doaj/js/doaj.facetview.theme.js]: note to self remove this before merge

File: portality/static/doaj/js/doaj.facetview.theme.js [hunk 2]
```diff
@@ -133,22 +245,32 @@ function doajScrollTop(options, context) {
...
+    doajToggleAbstract(options, context);
+
+    // Update the widget options & generated text
+    if (options.sharesave_link) {
+        var widget_text = doajGenFixedQueryWidget(options);
+        $('.facetview_embedwidget', context).val(widget_text);
+    }
+}
+
+function doajFixedQueryWidgetPostRender(options, context) {
```
Comment (emanuil-tolev) [inline portality/static/doaj/js/doaj.facetview.theme.js]: :+1: very nice use of the 2 different functions to achieve different results with normal integration and the widget. fv2 seems to have gotten a lot of code structure right to allow custom f() to be written in this particular manner. Don't know what you're missing till you don't have it (with fv1).

File: portality/templates/doaj/widgets.html [hunk 3]
```diff
@@ -3,50 +3,187 @@
...
+    sort_field: 'index.unpunctitle.exact',         // Field to order results by
+    sort_direction:  'asc',                        // Direction of sort "asc" | "desc"
+    search_operator : 'AND',                       // Which sort operator to use "AND" | "OR"
+    search_type: 'journal',                        // Which type to search upon (omit for both) "article" | "journal"
+    page_size : 5,                                 // How many results to show per widget page
+    page_from : 0                                  // Which result to initially start from
+    }
+</script>
+<script src="/static/widget/fixed_query.js" type="text/javascript"></script>
+{% endblock extra_js_bottom %}
```
Comment (emanuil-tolev) [inline portality/templates/doaj/widgets.html]: very nice docs, thank you @Steven-Eardley

**Comment by emanuil-tolev** [comment]
This is ready to merge, I removed just one diagnostic logging line. I will merge when it's built.

**Comment by Steven-Eardley** [comment]
@emanuil-tolev it looks like you forgot to un-localhost the widget. See PR message above.

**Comment by emanuil-tolev** [comment]
Ah so I did, will do that now with multiple HTML files to integrate like the other one. For now nothing terrible will happen except it won't work on test :).

**Comment by Steven-Eardley** [comment]
It won't work on test anyway - it permanently retrieves the html from doaj.org. So only when it's deployed live will it appear on any system!

**Comment by emanuil-tolev** [comment]
True. Might as well do that then.

**Pull request merged**